### PR TITLE
ROX-17931: refactor resolver to use uniqueue

### PR DESCRIPTION
**Pull request by lvalerom**
## Description

Currently two resources could trigger the same deployment reprocess. The current implementation will process the same deployment twice; this is not necessary. We could instead aggregate these two deployment references and process the deployment just once.

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

* [x] CI
* [x] Sensor integration tests pass locally
* [x] Unit tests pass locally

### CODE REVIEW

File: pkg/uniqueue/uniqueue_test.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+var (
+	queueSize      = 5
+	frontValue     = 0
+	backValue      = 3
+	numsToPush     = []int{1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2}
+	expectedSlice  = []int{frontValue, 1, 2, backValue}
+	expectedMap    = map[int]struct{}{frontValue: {}, 1: {}, 2: {}, backValue: {}}
+	defaultTimeout = 1 * time.Second
+)
+
```
Comment (vikin91) [inline pkg/uniqueue/uniqueue_test.go:24]: Let's add a test for possible goroutine leaks. Here an example:

```go
import 	"go.uber.org/goleak"


func assertNoGoroutineLeaks(t *testing.T) {
	goleak.VerifyNone(t,
		// Ignore a known leak: https://github.com/DataDog/dd-trace-go/issues/1469
		goleak.IgnoreTopFunction("github.com/golang/glog.(*fileSink).flushDaemon"),
	)
}

func (s *NodeInventoryHandlerTestSuite) TearDownTest() {
	assertNoGoroutineLeaks(s.T())
}
```

File: pkg/uniqueue/uniqueue.go [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+func (q *UniQueue[T]) maybeAddToQueue(el T) bool {
+	q.mu.Lock()
+	defer q.mu.Unlock()
+	if _, alreadyInQueue := q.inQueue[el]; alreadyInQueue {
+		return false
+	}
+	q.inQueue[el] = struct{}{}
+	return true
+}
```
Comment (vikin91) [inline pkg/uniqueue/uniqueue.go:123]: This would be a great place to add a Prometheus metric for counting how many duplicates we have avoided.

**Review by vikin91 — COMMENTED** [review]
Adding two things that I wrote down after the sync review.

**Comment by lvalerom** [comment]
Closing in favor of #10269

**Comment by lvalerom** [comment]
Closing in favor of #10269

**Pull request closed without merging**